numba/config.py

```diff
@@ -74,6 +81,22 @@ def reset(self):
 
     def update(self, force=False):
         new_environ = {}
+
+        # first check of a .numba_config.yaml and use values from that
+        if os.path.exists(_config_fname) and os.path.isfile(_config_fname):
+            if not _HAVE_YAML:
+                msg = ("A Numba config file if found but YAML parsing "
+                       "capabilities appear to be missing."
+                       "To use this feature please install `pyyaml`. e.g. "
+                       "`conda install pyyaml`.")
+                warnings.warn(msg)
+            else:
+                with open(_config_fname, 'rt') as f:
+                    y_conf = yaml.load(f)
+                for k, v in y_conf.items():
```

Comment:
`y_conf` can be `None` is the file is empty

Reply:
thanks :) this needs tests added before merge!